{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "535b109e-0f68-472a-91e4-cbd177b2b709",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "\"package 'msigdbr' was built under R version 4.2.3\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 x 18</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gs_cat</th><th scope=col>gs_subcat</th><th scope=col>gs_name</th><th scope=col>gene_symbol</th><th scope=col>entrez_gene</th><th scope=col>ensembl_gene</th><th scope=col>human_gene_symbol</th><th scope=col>human_entrez_gene</th><th scope=col>human_ensembl_gene</th><th scope=col>gs_id</th><th scope=col>gs_pmid</th><th scope=col>gs_geoid</th><th scope=col>gs_exact_source</th><th scope=col>gs_url</th><th scope=col>gs_description</th><th scope=col>taxon_id</th><th scope=col>ortholog_sources</th><th scope=col>num_ortholog_sources</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>C3</td><td>MIR:MIR_Legacy</td><td>AAACCAC_MIR140</td><td>Abcc4   </td><td>239273</td><td>ENSMUSG00000032849</td><td>ABCC4   </td><td>10257</td><td>ENSG00000125257</td><td>M12609</td><td></td><td></td><td></td><td></td><td>Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</td><td>10090</td><td>Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther                         </td><td> 9</td></tr>\n",
       "\t<tr><td>C3</td><td>MIR:MIR_Legacy</td><td>AAACCAC_MIR140</td><td>Abraxas2</td><td>109359</td><td>ENSMUSG00000030965</td><td>ABRAXAS2</td><td>23172</td><td>ENSG00000165660</td><td>M12609</td><td></td><td></td><td></td><td></td><td>Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</td><td>10090</td><td>EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</td><td>12</td></tr>\n",
       "\t<tr><td>C3</td><td>MIR:MIR_Legacy</td><td>AAACCAC_MIR140</td><td>Actn4   </td><td> 60595</td><td>ENSMUSG00000054808</td><td>ACTN4   </td><td>   81</td><td>ENSG00000130402</td><td>M12609</td><td></td><td></td><td></td><td></td><td>Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</td><td>10090</td><td>EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</td><td>12</td></tr>\n",
       "\t<tr><td>C3</td><td>MIR:MIR_Legacy</td><td>AAACCAC_MIR140</td><td>Acvr1   </td><td> 11477</td><td>ENSMUSG00000026836</td><td>ACVR1   </td><td>   90</td><td>ENSG00000115170</td><td>M12609</td><td></td><td></td><td></td><td></td><td>Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</td><td>10090</td><td>EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</td><td>12</td></tr>\n",
       "\t<tr><td>C3</td><td>MIR:MIR_Legacy</td><td>AAACCAC_MIR140</td><td>Adam9   </td><td> 11502</td><td>ENSMUSG00000031555</td><td>ADAM9   </td><td> 8754</td><td>ENSG00000168615</td><td>M12609</td><td></td><td></td><td></td><td></td><td>Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</td><td>10090</td><td>EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam          </td><td>11</td></tr>\n",
       "\t<tr><td>C3</td><td>MIR:MIR_Legacy</td><td>AAACCAC_MIR140</td><td>Adamts5 </td><td> 23794</td><td>ENSMUSG00000022894</td><td>ADAMTS5 </td><td>11096</td><td>ENSG00000154736</td><td>M12609</td><td></td><td></td><td></td><td></td><td>Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</td><td>10090</td><td>EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam          </td><td>11</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 x 18\n",
       "\\begin{tabular}{llllllllllllllllll}\n",
       " gs\\_cat & gs\\_subcat & gs\\_name & gene\\_symbol & entrez\\_gene & ensembl\\_gene & human\\_gene\\_symbol & human\\_entrez\\_gene & human\\_ensembl\\_gene & gs\\_id & gs\\_pmid & gs\\_geoid & gs\\_exact\\_source & gs\\_url & gs\\_description & taxon\\_id & ortholog\\_sources & num\\_ortholog\\_sources\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t C3 & MIR:MIR\\_Legacy & AAACCAC\\_MIR140 & Abcc4    & 239273 & ENSMUSG00000032849 & ABCC4    & 10257 & ENSG00000125257 & M12609 &  &  &  &  & Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase). & 10090 & Ensembl\\textbar{}HGNC\\textbar{}HomoloGene\\textbar{}Inparanoid\\textbar{}NCBI\\textbar{}OMA\\textbar{}OrthoDB\\textbar{}OrthoMCL\\textbar{}Panther                          &  9\\\\\n",
       "\t C3 & MIR:MIR\\_Legacy & AAACCAC\\_MIR140 & Abraxas2 & 109359 & ENSMUSG00000030965 & ABRAXAS2 & 23172 & ENSG00000165660 & M12609 &  &  &  &  & Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase). & 10090 & EggNOG\\textbar{}Ensembl\\textbar{}HGNC\\textbar{}HomoloGene\\textbar{}Inparanoid\\textbar{}NCBI\\textbar{}OMA\\textbar{}OrthoDB\\textbar{}OrthoMCL\\textbar{}Panther\\textbar{}PhylomeDB\\textbar{}Treefam & 12\\\\\n",
       "\t C3 & MIR:MIR\\_Legacy & AAACCAC\\_MIR140 & Actn4    &  60595 & ENSMUSG00000054808 & ACTN4    &    81 & ENSG00000130402 & M12609 &  &  &  &  & Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase). & 10090 & EggNOG\\textbar{}Ensembl\\textbar{}HGNC\\textbar{}HomoloGene\\textbar{}Inparanoid\\textbar{}NCBI\\textbar{}OMA\\textbar{}OrthoDB\\textbar{}OrthoMCL\\textbar{}Panther\\textbar{}PhylomeDB\\textbar{}Treefam & 12\\\\\n",
       "\t C3 & MIR:MIR\\_Legacy & AAACCAC\\_MIR140 & Acvr1    &  11477 & ENSMUSG00000026836 & ACVR1    &    90 & ENSG00000115170 & M12609 &  &  &  &  & Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase). & 10090 & EggNOG\\textbar{}Ensembl\\textbar{}HGNC\\textbar{}HomoloGene\\textbar{}Inparanoid\\textbar{}NCBI\\textbar{}OMA\\textbar{}OrthoDB\\textbar{}OrthoMCL\\textbar{}Panther\\textbar{}PhylomeDB\\textbar{}Treefam & 12\\\\\n",
       "\t C3 & MIR:MIR\\_Legacy & AAACCAC\\_MIR140 & Adam9    &  11502 & ENSMUSG00000031555 & ADAM9    &  8754 & ENSG00000168615 & M12609 &  &  &  &  & Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase). & 10090 & EggNOG\\textbar{}Ensembl\\textbar{}HGNC\\textbar{}HomoloGene\\textbar{}Inparanoid\\textbar{}NCBI\\textbar{}OMA\\textbar{}OrthoDB\\textbar{}OrthoMCL\\textbar{}Panther\\textbar{}Treefam           & 11\\\\\n",
       "\t C3 & MIR:MIR\\_Legacy & AAACCAC\\_MIR140 & Adamts5  &  23794 & ENSMUSG00000022894 & ADAMTS5  & 11096 & ENSG00000154736 & M12609 &  &  &  &  & Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase). & 10090 & EggNOG\\textbar{}Ensembl\\textbar{}HGNC\\textbar{}HomoloGene\\textbar{}Inparanoid\\textbar{}NCBI\\textbar{}OMA\\textbar{}OrthoDB\\textbar{}OrthoMCL\\textbar{}Panther\\textbar{}Treefam           & 11\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 x 18\n",
       "\n",
       "| gs_cat &lt;chr&gt; | gs_subcat &lt;chr&gt; | gs_name &lt;chr&gt; | gene_symbol &lt;chr&gt; | entrez_gene &lt;int&gt; | ensembl_gene &lt;chr&gt; | human_gene_symbol &lt;chr&gt; | human_entrez_gene &lt;int&gt; | human_ensembl_gene &lt;chr&gt; | gs_id &lt;chr&gt; | gs_pmid &lt;chr&gt; | gs_geoid &lt;chr&gt; | gs_exact_source &lt;chr&gt; | gs_url &lt;chr&gt; | gs_description &lt;chr&gt; | taxon_id &lt;int&gt; | ortholog_sources &lt;chr&gt; | num_ortholog_sources &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| C3 | MIR:MIR_Legacy | AAACCAC_MIR140 | Abcc4    | 239273 | ENSMUSG00000032849 | ABCC4    | 10257 | ENSG00000125257 | M12609 | <!----> | <!----> | <!----> | <!----> | Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase). | 10090 | Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther                          |  9 |\n",
       "| C3 | MIR:MIR_Legacy | AAACCAC_MIR140 | Abraxas2 | 109359 | ENSMUSG00000030965 | ABRAXAS2 | 23172 | ENSG00000165660 | M12609 | <!----> | <!----> | <!----> | <!----> | Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase). | 10090 | EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam | 12 |\n",
       "| C3 | MIR:MIR_Legacy | AAACCAC_MIR140 | Actn4    |  60595 | ENSMUSG00000054808 | ACTN4    |    81 | ENSG00000130402 | M12609 | <!----> | <!----> | <!----> | <!----> | Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase). | 10090 | EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam | 12 |\n",
       "| C3 | MIR:MIR_Legacy | AAACCAC_MIR140 | Acvr1    |  11477 | ENSMUSG00000026836 | ACVR1    |    90 | ENSG00000115170 | M12609 | <!----> | <!----> | <!----> | <!----> | Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase). | 10090 | EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam | 12 |\n",
       "| C3 | MIR:MIR_Legacy | AAACCAC_MIR140 | Adam9    |  11502 | ENSMUSG00000031555 | ADAM9    |  8754 | ENSG00000168615 | M12609 | <!----> | <!----> | <!----> | <!----> | Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase). | 10090 | EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam           | 11 |\n",
       "| C3 | MIR:MIR_Legacy | AAACCAC_MIR140 | Adamts5  |  23794 | ENSMUSG00000022894 | ADAMTS5  | 11096 | ENSG00000154736 | M12609 | <!----> | <!----> | <!----> | <!----> | Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase). | 10090 | EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam           | 11 |\n",
       "\n"
      ],
      "text/plain": [
       "  gs_cat gs_subcat      gs_name        gene_symbol entrez_gene\n",
       "1 C3     MIR:MIR_Legacy AAACCAC_MIR140 Abcc4       239273     \n",
       "2 C3     MIR:MIR_Legacy AAACCAC_MIR140 Abraxas2    109359     \n",
       "3 C3     MIR:MIR_Legacy AAACCAC_MIR140 Actn4        60595     \n",
       "4 C3     MIR:MIR_Legacy AAACCAC_MIR140 Acvr1        11477     \n",
       "5 C3     MIR:MIR_Legacy AAACCAC_MIR140 Adam9        11502     \n",
       "6 C3     MIR:MIR_Legacy AAACCAC_MIR140 Adamts5      23794     \n",
       "  ensembl_gene       human_gene_symbol human_entrez_gene human_ensembl_gene\n",
       "1 ENSMUSG00000032849 ABCC4             10257             ENSG00000125257   \n",
       "2 ENSMUSG00000030965 ABRAXAS2          23172             ENSG00000165660   \n",
       "3 ENSMUSG00000054808 ACTN4                81             ENSG00000130402   \n",
       "4 ENSMUSG00000026836 ACVR1                90             ENSG00000115170   \n",
       "5 ENSMUSG00000031555 ADAM9              8754             ENSG00000168615   \n",
       "6 ENSMUSG00000022894 ADAMTS5           11096             ENSG00000154736   \n",
       "  gs_id  gs_pmid gs_geoid gs_exact_source gs_url\n",
       "1 M12609                                        \n",
       "2 M12609                                        \n",
       "3 M12609                                        \n",
       "4 M12609                                        \n",
       "5 M12609                                        \n",
       "6 M12609                                        \n",
       "  gs_description                                                                                                                                                                                        \n",
       "1 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).\n",
       "2 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).\n",
       "3 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).\n",
       "4 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).\n",
       "5 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).\n",
       "6 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).\n",
       "  taxon_id\n",
       "1 10090   \n",
       "2 10090   \n",
       "3 10090   \n",
       "4 10090   \n",
       "5 10090   \n",
       "6 10090   \n",
       "  ortholog_sources                                                                             \n",
       "1 Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther                         \n",
       "2 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam\n",
       "3 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam\n",
       "4 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam\n",
       "5 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam          \n",
       "6 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam          \n",
       "  num_ortholog_sources\n",
       "1  9                  \n",
       "2 12                  \n",
       "3 12                  \n",
       "4 12                  \n",
       "5 11                  \n",
       "6 11                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 30 x 1</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_symbol</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Acly  </td></tr>\n",
       "\t<tr><td>Aco1  </td></tr>\n",
       "\t<tr><td>Aco2  </td></tr>\n",
       "\t<tr><td>Cs    </td></tr>\n",
       "\t<tr><td>Dlat  </td></tr>\n",
       "\t<tr><td>Dld   </td></tr>\n",
       "\t<tr><td>Dlst  </td></tr>\n",
       "\t<tr><td>Fh1   </td></tr>\n",
       "\t<tr><td>Idh1  </td></tr>\n",
       "\t<tr><td>Idh2  </td></tr>\n",
       "\t<tr><td>Idh3a </td></tr>\n",
       "\t<tr><td>Idh3b </td></tr>\n",
       "\t<tr><td>Idh3g </td></tr>\n",
       "\t<tr><td>Mdh1  </td></tr>\n",
       "\t<tr><td>Mdh2  </td></tr>\n",
       "\t<tr><td>Ogdh  </td></tr>\n",
       "\t<tr><td>Ogdhl </td></tr>\n",
       "\t<tr><td>Pcx   </td></tr>\n",
       "\t<tr><td>Pck1  </td></tr>\n",
       "\t<tr><td>Pck2  </td></tr>\n",
       "\t<tr><td>Pdha1 </td></tr>\n",
       "\t<tr><td>Pdha2 </td></tr>\n",
       "\t<tr><td>Pdhb  </td></tr>\n",
       "\t<tr><td>Sdha  </td></tr>\n",
       "\t<tr><td>Sdhb  </td></tr>\n",
       "\t<tr><td>Sdhc  </td></tr>\n",
       "\t<tr><td>Sdhd  </td></tr>\n",
       "\t<tr><td>Sucla2</td></tr>\n",
       "\t<tr><td>Suclg1</td></tr>\n",
       "\t<tr><td>Suclg2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 30 x 1\n",
       "\\begin{tabular}{l}\n",
       " gene\\_symbol\\\\\n",
       " <chr>\\\\\n",
       "\\hline\n",
       "\t Acly  \\\\\n",
       "\t Aco1  \\\\\n",
       "\t Aco2  \\\\\n",
       "\t Cs    \\\\\n",
       "\t Dlat  \\\\\n",
       "\t Dld   \\\\\n",
       "\t Dlst  \\\\\n",
       "\t Fh1   \\\\\n",
       "\t Idh1  \\\\\n",
       "\t Idh2  \\\\\n",
       "\t Idh3a \\\\\n",
       "\t Idh3b \\\\\n",
       "\t Idh3g \\\\\n",
       "\t Mdh1  \\\\\n",
       "\t Mdh2  \\\\\n",
       "\t Ogdh  \\\\\n",
       "\t Ogdhl \\\\\n",
       "\t Pcx   \\\\\n",
       "\t Pck1  \\\\\n",
       "\t Pck2  \\\\\n",
       "\t Pdha1 \\\\\n",
       "\t Pdha2 \\\\\n",
       "\t Pdhb  \\\\\n",
       "\t Sdha  \\\\\n",
       "\t Sdhb  \\\\\n",
       "\t Sdhc  \\\\\n",
       "\t Sdhd  \\\\\n",
       "\t Sucla2\\\\\n",
       "\t Suclg1\\\\\n",
       "\t Suclg2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 30 x 1\n",
       "\n",
       "| gene_symbol &lt;chr&gt; |\n",
       "|---|\n",
       "| Acly   |\n",
       "| Aco1   |\n",
       "| Aco2   |\n",
       "| Cs     |\n",
       "| Dlat   |\n",
       "| Dld    |\n",
       "| Dlst   |\n",
       "| Fh1    |\n",
       "| Idh1   |\n",
       "| Idh2   |\n",
       "| Idh3a  |\n",
       "| Idh3b  |\n",
       "| Idh3g  |\n",
       "| Mdh1   |\n",
       "| Mdh2   |\n",
       "| Ogdh   |\n",
       "| Ogdhl  |\n",
       "| Pcx    |\n",
       "| Pck1   |\n",
       "| Pck2   |\n",
       "| Pdha1  |\n",
       "| Pdha2  |\n",
       "| Pdhb   |\n",
       "| Sdha   |\n",
       "| Sdhb   |\n",
       "| Sdhc   |\n",
       "| Sdhd   |\n",
       "| Sucla2 |\n",
       "| Suclg1 |\n",
       "| Suclg2 |\n",
       "\n"
      ],
      "text/plain": [
       "   gene_symbol\n",
       "1  Acly       \n",
       "2  Aco1       \n",
       "3  Aco2       \n",
       "4  Cs         \n",
       "5  Dlat       \n",
       "6  Dld        \n",
       "7  Dlst       \n",
       "8  Fh1        \n",
       "9  Idh1       \n",
       "10 Idh2       \n",
       "11 Idh3a      \n",
       "12 Idh3b      \n",
       "13 Idh3g      \n",
       "14 Mdh1       \n",
       "15 Mdh2       \n",
       "16 Ogdh       \n",
       "17 Ogdhl      \n",
       "18 Pcx        \n",
       "19 Pck1       \n",
       "20 Pck2       \n",
       "21 Pdha1      \n",
       "22 Pdha2      \n",
       "23 Pdhb       \n",
       "24 Sdha       \n",
       "25 Sdhb       \n",
       "26 Sdhc       \n",
       "27 Sdhd       \n",
       "28 Sucla2     \n",
       "29 Suclg1     \n",
       "30 Suclg2     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 x 7</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Pathway</th><th scope=col>NAME</th><th scope=col>CLASS1</th><th scope=col>CLASS2</th><th scope=col>PATHWAY_MAP</th><th scope=col>COMPOUND</th><th scope=col>COMPOUND_DESCRIPTION</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>mmu00020</td><td>Citrate cycle (TCA cycle) - Mus musculus (house mouse)</td><td>Metabolism</td><td>Carbohydrate metabolism</td><td>Citrate cycle (TCA cycle)</td><td>C00022</td><td>Pyruvate</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 x 7\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & Pathway & NAME & CLASS1 & CLASS2 & PATHWAY\\_MAP & COMPOUND & COMPOUND\\_DESCRIPTION\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & mmu00020 & Citrate cycle (TCA cycle) - Mus musculus (house mouse) & Metabolism & Carbohydrate metabolism & Citrate cycle (TCA cycle) & C00022 & Pyruvate\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 x 7\n",
       "\n",
       "| <!--/--> | Pathway &lt;chr&gt; | NAME &lt;chr&gt; | CLASS1 &lt;chr&gt; | CLASS2 &lt;chr&gt; | PATHWAY_MAP &lt;chr&gt; | COMPOUND &lt;chr&gt; | COMPOUND_DESCRIPTION &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| 1 | mmu00020 | Citrate cycle (TCA cycle) - Mus musculus (house mouse) | Metabolism | Carbohydrate metabolism | Citrate cycle (TCA cycle) | C00022 | Pyruvate |\n",
       "\n"
      ],
      "text/plain": [
       "  Pathway  NAME                                                   CLASS1    \n",
       "1 mmu00020 Citrate cycle (TCA cycle) - Mus musculus (house mouse) Metabolism\n",
       "  CLASS2                  PATHWAY_MAP               COMPOUND\n",
       "1 Carbohydrate metabolism Citrate cycle (TCA cycle) C00022  \n",
       "  COMPOUND_DESCRIPTION\n",
       "1 Pyruvate            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'C00026'</li><li>'C00042'</li><li>'C00068'</li><li>'C00122'</li><li>'C00149'</li><li>'C00158'</li><li>'C00417'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'C00026'\n",
       "\\item 'C00042'\n",
       "\\item 'C00068'\n",
       "\\item 'C00122'\n",
       "\\item 'C00149'\n",
       "\\item 'C00158'\n",
       "\\item 'C00417'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'C00026'\n",
       "2. 'C00042'\n",
       "3. 'C00068'\n",
       "4. 'C00122'\n",
       "5. 'C00149'\n",
       "6. 'C00158'\n",
       "7. 'C00417'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"C00026\" \"C00042\" \"C00068\" \"C00122\" \"C00149\" \"C00158\" \"C00417\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 7 x 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Query</th><th scope=col>KEGG</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Citric acid            </td><td>C00158</td></tr>\n",
       "\t<tr><td>L-Malate               </td><td>C00149</td></tr>\n",
       "\t<tr><td>Succinic acid          </td><td>C00042</td></tr>\n",
       "\t<tr><td>alpha-Ketoglutaric acid</td><td>C00026</td></tr>\n",
       "\t<tr><td>TPP                    </td><td>C00068</td></tr>\n",
       "\t<tr><td>cis-Aconitic acid      </td><td>C00417</td></tr>\n",
       "\t<tr><td>Fumaric acid           </td><td>C00122</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 7 x 2\n",
       "\\begin{tabular}{ll}\n",
       " Query & KEGG\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t Citric acid             & C00158\\\\\n",
       "\t L-Malate                & C00149\\\\\n",
       "\t Succinic acid           & C00042\\\\\n",
       "\t alpha-Ketoglutaric acid & C00026\\\\\n",
       "\t TPP                     & C00068\\\\\n",
       "\t cis-Aconitic acid       & C00417\\\\\n",
       "\t Fumaric acid            & C00122\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 7 x 2\n",
       "\n",
       "| Query &lt;chr&gt; | KEGG &lt;chr&gt; |\n",
       "|---|---|\n",
       "| Citric acid             | C00158 |\n",
       "| L-Malate                | C00149 |\n",
       "| Succinic acid           | C00042 |\n",
       "| alpha-Ketoglutaric acid | C00026 |\n",
       "| TPP                     | C00068 |\n",
       "| cis-Aconitic acid       | C00417 |\n",
       "| Fumaric acid            | C00122 |\n",
       "\n"
      ],
      "text/plain": [
       "  Query                   KEGG  \n",
       "1 Citric acid             C00158\n",
       "2 L-Malate                C00149\n",
       "3 Succinic acid           C00042\n",
       "4 alpha-Ketoglutaric acid C00026\n",
       "5 TPP                     C00068\n",
       "6 cis-Aconitic acid       C00417\n",
       "7 Fumaric acid            C00122"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "setwd('/import/linhao/')\n",
    "\n",
    "library(msigdbr)\n",
    "\n",
    "m_df = msigdbr(species = \"Mus musculus\")\n",
    "head(m_df)\n",
    "\n",
    "m_df = msigdbr(species = \"Mus musculus\", category = \"C2\")\n",
    "\n",
    "m_df = msigdbr(species = \"Mus musculus\", category = \"C2\", subcategory = \"CP:KEGG\")\n",
    "\n",
    "m_df %>%  dplyr::filter(grepl('TCA', gs_name)) -> a\n",
    "\n",
    "a[,4]\n",
    "\n",
    "meta = read.csv('./mete.csv',header =T)\n",
    "\n",
    "kegg = read.csv('/import/194.data/data/workdirV6/KEGG_COMPOUND_PATHWAY_mmu.csv')\n",
    "\n",
    "kegg %>%  dplyr::filter(grepl('TCA', PATHWAY_MAP)) -> kegg2\n",
    "\n",
    "head(kegg2,1)\n",
    "\n",
    "intersect(kegg2$COMPOUND,meta$KEGG)\n",
    "\n",
    "\n",
    "\n",
    "meta %>%  dplyr::filter(KEGG %in% intersect(kegg2$COMPOUND,meta$KEGG) )\n",
    "\n",
    "meta %>%  dplyr::filter(KEGG %in% intersect(kegg2$COMPOUND,meta$KEGG) ) -> meta2\n",
    "\n",
    "data = read.csv('./pos.neg.csv')\n",
    "\n",
    "data %>%  dplyr::filter(Name %in% meta2$Query ) -> RES\n",
    "\n",
    "RES$log2FC = -RES$log2FC\n",
    "\n",
    "write.csv(RES,file = 'res.csv',row.names =F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34d9bfbe-9e00-4013-a673-080db80657b6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
